src/workflow-engine/charms/temporal-worker/oci_factory/activities/find_images_to_update.py

```diff
@@ -205,6 +206,10 @@ def find_released_revisions(releases_json: dict) -> list:
                     release_to[to_track]["end-of-life"] = releases[to_track][
                         "end-of-life"
                     ]
+                    if release_to[to_track]["end-of-life"] < datetime.now(
+                        timezone.utc
+                    ).strftime("%Y-%m-%dT%H:%M:%SZ"):
+                        del release_to[to_track]
```

Comment:
this is over complicating it. Just

```
if releases[to_track]["end-of-life"] < datetime.now...:
    # skip it, with logs

release_to[to_track]["end-of-life"] = releases[to_track]["end-of-life"] 
```

so, no need to `del`, and as above, you shouldn't update it EOL has passed. The logic should be exactly the same as for other jobs